pandas/core/categorical.py

```diff
@@ -896,11 +905,33 @@ def rename_categories(self, new_categories, inplace=False):
         remove_categories
         remove_unused_categories
         set_categories
+
+        Examples
+        --------
+        >>> c = Categorical(['a', 'a', 'b'])
+        >>> c.rename_categories([0, 1])
+        [0, 0, 1]
+        Categories (2, int64): [0, 1]
+
+        For dict-like ``new_categories``, extra keys are ignored and
+        categories not in the dictionary are passed through
+
+        >>> c.rename_categories({'a': 'A', 'c': 'C'})
+        [A, A, b]
+        Categories (2, object): [A, b]
+
+        Series are considered list-like here, so the *values* are used
+        instead of the *index*
```

Comment:
Do we actually want this behaviour? 
Eg for `Series.rename`, a Series is seen as a dict-like ...

Reply:
Not sure. It’ll be a backwards incompatible change if we don’t treat Series as arrays so I think we should at least do this for now, maybe with a warning.

Reply:
I agree, I think this is fine.

Reply:
@jreback which part to you agree with? Warning that it'll change to dict-like in the future?

Reply:
no I agree the current behavior is correct. we handle list-like the same. no warning is needed as this is expected.

Reply:
actually, looking at this again.

a Series should be just like a dict. This is a perf issue yes? 
do this. (once ``Index.map`` works we could simplify a bit)
```
In [12]: cat = pd.Categorical(['a', 'b', 'c', 'd'])
    ...: res = cat.rename_categories(pd.Series({'a': 4, 'b': 3, 'c': 2, 'd': 1}))
    ...: 
    ...: 

In [13]: cat
Out[13]: 
[a, b, c, d]
Categories (4, object): [a, b, c, d]

In [14]: res
Out[14]: 
[4, 3, 2, 1]
Categories (4, int64): [4, 3, 2, 1]

In [17]: pd.Series(cat.categories).map({'a': 4, 'b': 3, 'c': 2, 'd': 1}).values
Out[17]: array([4, 3, 2, 1])

In [19]: pd.Series(cat.categories).map({'a': 4, 'b': 3, 'c': 2, 'd': 1}).values
Out[19]: array([4, 3, 2, 1])

```

Reply:
please revert the warning

Reply:
Can you explain? 
If we go for "Series -> dict-like" behaviour, this is a breaking change, and we need to use a warning for that.